src/compressed_tensors/compressors/model_compressors/model_compressor.py

```diff
@@ -515,11 +541,60 @@ def _replace_weights(self, dense_weight_generator, model: Module):
         :param model: The model whose weights are to be updated.
         """
         for name, data in tqdm(dense_weight_generator, desc="Decompressing model"):
+
             split_name = name.split(".")
             prefix, param_name = ".".join(split_name[:-1]), split_name[-1]
             module = operator.attrgetter(prefix)(model)
-            if hasattr(module, param_name):
-                update_parameter_data(module, data, param_name)
+
+            params_device = next(module.parameters()).device
+            device = "cpu" if has_offloaded_params(module) else params_device
+            delattr(module, param_name)
+            requires_grad = data.dtype in (torch.float16, torch.float32, torch.bfloat16)
```

Comment:
Not sure if there's a reason we want to be explicit here, but you might be able to do this more generically
```suggestion
            requires_grad = torch.is_floating_point(data)
```

Reply:
https://github.com/pytorch/pytorch/blob/0f8613bf5cbdd7a2af5c46e6fa1adda35c69db8d/torch/csrc/autograd/variable.h#L270

https://github.com/pytorch/pytorch/blob/0f8613bf5cbdd7a2af5c46e6fa1adda35c69db8d/torch/csrc/autograd/variable.h#L49

Technically, it should be `requires_grad = data.dtype.is_floating_point or data.dtype.is_complex`

Reply:
But I don't think we technically want gradients to be enabled when set to fp8?

Reply:
I am going to keep the explicit list for the time being.